node/rpc/src/lib.rs

```diff
@@ -1,20 +1,38 @@
 // Copyright 2020 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+#[macro_use]
+extern crate lazy_static;
+
+mod auth_api;
 mod chain_api;
 mod mpool_api;
 mod sync_api;
 mod wallet_api;
 
 use async_std::sync::{RwLock, Sender};
+use auth::{has_perms, Error};
 use blockstore::BlockStore;
 use chain_sync::{BadBlockCache, SyncState};
 use forest_libp2p::NetworkMessage;
-use jsonrpc_v2::{Data, MapRouter, RequestObject, Server};
+use jsonrpc_v2::{Data, Error as JsonRpcError, ErrorLike, MapRouter, RequestObject, Server};
 use message_pool::{MessagePool, MpoolRpcProvider};
 use std::sync::Arc;
 use tide::{Request, Response, StatusCode};
+use utils::get_home_dir;
 use wallet::KeyStore;
+use wallet::PersistentKeyStore;
+
+lazy_static! {
+    pub static ref WRITE_ACCESS: Vec<String> = vec![
+        "Filecoin.MpoolPush".to_string(),
+        "Filecoin.WalletNew".to_string(),
+        "Filecoin.WalletHas".to_string(),
+        "Filecoin.WalletList".to_string(),
+        "Filecoin.WalletDefaultAddress".to_string(),
+        "Filecoin.WalletList".to_string(),
+    ];
+}
```

Comment:
This does not have to be lazy static, can just be an array of static strings (no need for allocations for the vec or strings